Add kubeval docs (#2273)
operate-first/apps: docs/content/kubeval/updating_schema_store.md

# Updating schemas in schema store

Our `apps` repository uses CRDs for which schemas can't be found in default schema repository. This means that we have to use our own schema repository. For this we have used [script](https://github.com/sabre1041/k8s-manifest-validation/blob/main/scripts/build_schema.py) written by Andrew Block. This script generates schemas from OpenAPI specification using [openapi2jsonschema](https://github.com/instrumenta/openapi2jsonschema). We have also extended this script to generate `skip_kinds.txt` which contains names of the CRDs which don't have valid schemas and will not be validated in `kubeval` validation. Script can be found in [schema-store](https://github.com/operate-first/schema-store) repository.

> To use script you run have to login to cluster with `oc` and run following:

Reviewer: ```suggestion
To use this script you will have to login to the cluster with `oc` and run following:
```

Author: Resolved in latest commit.

```bash
python build_schema.py -u $(oc whoami --show-server) -t $(oc whoami -t) -s STRICT
```

> Script generates folder called `openshift-json-schema/master-standalone-strict` which contains schemas for all CRDs which are present in the cluster.

Reviewer: ```suggestion
This script generates a folder called `openshift-json-schema/master-standalone-strict` which contains schemas for all CRDs which are present in this cluster.
```

Author: Resolved in latest commit.

> Script is then used in github actions, where we [run cronjob](https://github.com/operate-first/schema-store/blob/main/.github/workflows/update-schemas.yaml), which generates schemas from each cluster and joins them together. In the end if the PR doesn't exist it is created with new changes or if there is opened PR changes are updated.

Reviewer: ```suggestion
This script is then used in github actions, where we [run cronjob](https://github.com/operate-first/schema-store/blob/main/.github/workflows/update-schemas.yaml), which generates schemas from each cluster and joins them together. In the end if the PR doesn't exist it is created with new changes or if there is opened PR changes are updated.
```

Author: Resolved in latest commit.
